### PR TITLE
New operability for CuPy arrays

### PR DESCRIPTION
**Pull request by dkazanc**
fast reconstruction using FBP and iterative methods with device-to-device capability.

### CODE REVIEW

File: tomobar/supp/astraOP.py [hunk 1]
```diff
@@ -10,6 +10,19 @@
 """
 import numpy as np
 
+gpu_enabled = False
+try:
+    import cupy as xp
+    try:
+        xp.cuda.Device(0).compute_capability
+        gpu_enabled = True  # CuPy is installed and GPU is available
```
Comment (kalvdans) [inline tomobar/supp/astraOP.py:18]: unused constant, remove.

File: tomobar/supp/astraOP.py [hunk 2]
```diff
@@ -10,6 +10,19 @@
 """
 import numpy as np
 
+gpu_enabled = False
```
Comment (kalvdans) [inline tomobar/supp/astraOP.py:13]: unused constant, remove

Comment (dkazanc) [reply tomobar/supp/astraOP.py]: thanks, fixed

File: tomobar/supp/astraOP.py (hunk 1 above)
Comment (dkazanc) [reply tomobar/supp/astraOP.py]: thanks, fixed

File: tomobar/methodsCuPy.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+GPLv3 license (ASTRA toolbox)
+@author: Daniil Kazantsev: https://github.com/dkazanc
+"""
+
+import numpy as np
+try:
+    import cupy as cp
+    import cupyx
+except ImportError:
+    raise ImportError("CuPy package is required, please install")
```
[the remark this answers is not in the document]
Comment (dkazanc) [reply tomobar/methodsCuPy.py]: replaced it to just a print message, could be useful still?

[the remark this answers is not in the document]
Comment (kalvdans) [reply tomobar/methodsCuPy.py]: No, remove the try...except clause.

**Pull request merged**